File: app/src/main/java/com/davidbronn/fragmentrevision/fragment_states/BaseActivity.java
package com.davidbronn.fragmentrevision.fragment_states;

import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v4.app.FragmentTransaction;

public class BaseActivity extends AppCompatActivity {

    public void addFragment(int container, boolean backstack, Fragment fragment) {
        String tag = fragment.getClass().getName();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.add(container, fragment, tag);
        if (backstack) {
            transaction.addToBackStack("add"+tag);
        }
        transaction.commit();
    }

    public void replaceFragment(int container, boolean backstack, Fragment fragment) {
        String tag = fragment.getClass().getName();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(container, fragment, tag);
        if (backstack) {
            transaction.addToBackStack("replace"+tag);
        }
        transaction.commit();
    }
}

File: app/src/main/java/com/davidbronn/fragmentrevision/utils/BaseActivity.kt
package com.davidbronn.fragmentrevision.utils

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

open class BaseActivity : AppCompatActivity() {

    fun addFragment(container: Int, fragment: Fragment, keepInStack: Boolean = true) {
        val tag = fragment.javaClass.name
        val transaction = supportFragmentManager.beginTransaction()
        transaction.add(container, fragment, tag)
        if (keepInStack) {
            transaction.addToBackStack("add$tag")
        }
        transaction.commit()
    }

    fun replaceFragment(container: Int, fragment: Fragment, keepInStack: Boolean = true) {
        val tag = fragment.javaClass.name
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(container, fragment, tag)
        if (keepInStack) {
            transaction.addToBackStack("replace$tag")
        }
        transaction.commit()
    }
}
